extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -373,40 +369,45 @@ export class AmpStoryPage extends AMP.BaseElement {
   }
 
   /**
-   * Reads the storyNextUp param and sets it as the auto-advance-after attribute
-   * if this is the last page and there isn't a value set by the publisher.
+   * Reads the storyNextUp param if provided and sets the auto-advance-after
+   * attribute if there isn't a value set by the publisher. The
+   * auto-advance-after attribute will be set to the id of the first
+   * video if there is one, or the value of the storyNextUp param otherwise.
    * @private
    */
   maybeSetStoryNextUp_() {
     const autoAdvanceAttr = this.element.getAttribute('auto-advance-after');
+    // This is a private param used for testing, it may be changed
+    // or removed without notice.
     const storyNextUpParam = Services.viewerForDoc(this.element).getParam(
       'storyNextUp'
     );
-    if (
-      autoAdvanceAttr === null &&
-      storyNextUpParam !== null &&
-      this.isLastPage_()
-    ) {
+    if (autoAdvanceAttr !== null || storyNextUpParam === null) {
+      return;
+    }
+    const video = this.getFirstVideo_();
+    if (video !== null) {
       addAttributesToElement(
         this.element,
-        dict({'auto-advance-after': storyNextUpParam})
+        dict({'auto-advance-after': video.id})
       );
+      return;
     }
+    addAttributesToElement(
+      this.element,
+      dict({'auto-advance-after': storyNextUpParam})
+    );
   }
 
   /**
-   * Returns true if the page is the last amp-story-page in the amp-story.
-   * @return {boolean}
+   * Returns the first video in the amp-story-page if there is one, otherwise
+   * returns null.
+   * @return {?Element}
    * @private
    */
-  isLastPage_() {
-    if (this.isLastStoryPage_ === null) {
-      this.isLastStoryPage_ = matches(
-        this.element,
-        'amp-story-page:last-of-type'
-      );
-    }
-    return this.isLastStoryPage_;
+  getFirstVideo_() {
+    const videos = this.getAllVideos_();
```

Comment:
`getAllVideos_()` will return `<video` elements. The auto-advancement class expects an ID to an `amp-video` element.

You can use `this.getMediaBySelector_(Selectors.ALL_AMP_VIDEO)` and create a helper method like the `getAllVideos_()` one as it'd now be used multiple times.

Reply:
Done. Thanks for catching this.